common/ast/ast.go

```diff
@@ -100,7 +99,7 @@ func (a *AST) SetReference(id int64, r *ReferenceInfo) {
 
 // IsChecked returns whether the AST is type-checked.
 func (a *AST) IsChecked() bool {
-	return a != nil && a.checked
+	return a != nil && (len(a.TypeMap()) > 0 || len(a.ReferenceMap()) > 0)
```

Comment:
Would checking the type map sufficient here? Can there be an AST that has references but no types?

Reply:
You're right, I think you just need the type map